ui/tests/acceptance/secrets/backend/totp/key-test.js

```diff
@@ -0,0 +1,173 @@
+    await click('[data-test-radio="Other service"]');
+    await fillIn(GENERAL.inputByAttr('name'), keyName);
+    await fillIn(GENERAL.inputByAttr('issuer'), issuer);
+    await fillIn(GENERAL.inputByAttr('accountName'), accountName);
+    if (url) await fillIn(GENERAL.inputByAttr('url'), url);
+    if (key) await fillIn(GENERAL.inputByAttr('key'), key);
+    await click('[data-test-totp-create]');
+  };
+
+  hooks.beforeEach(async function () {
+    this.uid = uuidv4();
+    this.mountPath = `totp-${this.uid}`;
+    this.path = `totp-${this.uid}`;
+    this.keyName = 'totp-key';
+    this.issuer = 'totp-issuer';
+    this.accountName = 'totp-acount';
+    this.url =
+      'otpauth://totp/test-issuer:my-account?algorithm=SHA1&digits=6&issuer=test-issuer&period=30&secret=HICPOBIMFO4YYHFYX3QPVYUL2YEPVJKU';
+    this.key = 'VCUDXBWFQXUEAIJYXH5YB62D5WFNQXFA';
+    await authPage.login();
+    // Setup TOTP engine
+    await mountBackend('totp', this.mountPath);
+  });
+
+  test('it views a key via menu option', async function (assert) {
+    assert.strictEqual(
+      currentURL(),
+      `/vault/secrets/${this.path}/list`,
+      'After enabling totp secrets engine it navigates to keys list'
+    );
+
+    await click(SES.createSecret);
+    await createVaultKey(this.keyName, this.issuer, this.accountName);
+    await visit(`/vault/secrets/${this.path}`);
+    await click(GENERAL.menuTrigger);
+    await click(`${SELECTORS.menuItem('details')}`);
+
+    assert.dom('.title').hasText(`TOTP key ${this.keyName}`);
+    assert.dom('[data-test-totp-key-details]').exists();
+
+    assert.strictEqual(
+      currentURL(),
+      `/vault/secrets/${this.path}/show/${this.keyName}`,
+      'After clicking details option it navigates to key detail view'
+    );
+  });
+
+  test('it deletes a key via menu option', async function (assert) {
+    assert.strictEqual(
+      currentURL(),
+      `/vault/secrets/${this.path}/list`,
+      'After enabling totp secrets engine it navigates to keys list'
+    );
+
+    await click(SES.createSecret);
+    await createVaultKey(this.keyName, this.issuer, this.accountName);
+    await waitUntil(() => currentURL() === `/vault/secrets/${this.path}/show/${this.keyName}`);
+    await visit(`/vault/secrets/${this.path}`);
+    await click(GENERAL.menuTrigger);
+    await click(GENERAL.confirmTrigger);
+    await click(GENERAL.confirmButton);
+    assert.dom(SES.secretLink(this.keyName)).doesNotExist(`${this.keyName}: key is no longer in the list`);
+    assert.strictEqual(
+      flashMessage.latestMessage,
+      `${this.keyName} was successfully deleted.`,
+      'renders delete flash upon key creation'
+    );
+  });
+
+  test('it creates a key with Vault as the provider', async function (assert) {
+    assert.strictEqual(
+      currentURL(),
+      `/vault/secrets/${this.path}/list`,
+      'After enabling totp secrets engine it navigates to keys list'
+    );
```

Comment:
we already tested this in one test, so probably don't need to repeat this assertion